content/tutorials/build-apps/build-a-desktop-wallet-in-js.md

```diff
@@ -0,0 +1,189 @@
+
+4. Now that we have the HTML code, let's add the JavaScript code. In the `helpers` folder, create a new file named `submit-transaction.js` and copy the code written below to the file. In this file, we are using the [submit](https://xrpl.org/submit.html) method to submit the transaction to the XRPL. Before submitting every transaction needs to be signed by a wallet, click [here](https://xrpl.org/sign.html) to learn more about signing a transaction.
+
+{{ include_code("_code-samples/build-a-wallet/js/src/helpers/submit-transaction.js", language="js") }}
+
+
+1. Now back to the send-xrp.js file, copy the code written below to the file. In this piece of code we are first getting all the DOM elements from HTML and adding event listners to update & validate the fields based on the user input. Using `renderAvailableBalance` method we display the current available balance of the wallet. `validateAddress` function validates the user address, and the amount is validated using a regular expression. When all the fields are filled with correct inputs, we call the `submitTransaction` function to submit the transaction to the ledger.
+
+{{ include_code("_code-samples/build-a-wallet/js/src/send-xrp/send-xrp.js", language="js") }}
+
+### Step 5: Creating the Transactions Page
+
+Now that we have created the home page and the send XRP page, let's create the transactions page that will display the transaction history of the account.
+
+  - Account: The account that sent the transaction.
+  - Destination: The account that received the transaction.
+  - Amount: The amount of XRP sent in the transaction.
+  - Transaction Type: The type of transaction.
+  - Result: The result of the transaction.
+  - Link: A link to the transaction on the XRP Ledger Explorer.
+
+![Transactions Page Screenshot](img/js-wallet-transaction.png)
+
+1. Create a folder named `transaction-history` in the src directory.
+
+2. Add two files named `transaction-history.js` and `transaction-history.html`.
+
+3. In the `transaction-history.js` file, copy the code written below. To fetch the transactions from XRP Ledger we are using [account_tx](https://xrpl.org/account_tx.html) method. This method returns a list of transactions that affected the specified account. We are using the `limit` parameter to limit the number of transactions to 10 per page. We are also using the `marker` parameter to paginate through the transactions.
+
+{{ include_code("_code-samples/build-a-wallet/js/src/transaction-history/transaction-history.js", language="js") }}
+
+1. Copy the contents of the [transaction-history.html](_code-samples/build-a-wallet/js/src/transaction-history/transaction-history.html) file to your transaction-history.html file. In this file we are defining a table  which displays the mentioned fields mentioned above.
```

Comment:
Two things here:
1. Creating the file should happen in the same step as "copying this code"
2. The additional text giving context should be on a separate line from the step (Generally it's better to have the steps be focused on actions, with any surrounding text giving more information once they've completed the task)
```suggestion
2. Create a file named `transaction-history.js` and copy the code written below. 

{{ include_code("_code-samples/build-a-wallet/js/src/transaction-history/transaction-history.js", language="js") }}

This code uses [account_tx](https://xrpl.org/account_tx.html) to fetch transactions we've sent to and from this account. In order to get all the results, we're using the `marker` parameter to paginate through the incomplete list of transactions until we reach the end.

3. Create a file named "transaction-history.html" and copy the code from [transaction-history.html](_code-samples/build-a-wallet/js/src/transaction-history/transaction-history.html) into it. 

`transaction-history.html` defines a table which displays the fields mentioned above.
```